statsmodels/survey/summary_stats.py

```diff
@@ -0,0 +1,626 @@
+        """
+        Computes bootstrap replicate weight
+
+        Returns
+        -------
+        w : ndarray
+            Augmented weight
+        """
+        w = self.weights.copy()
+        clust_count = np.zeros(self.nclust)
+        for s in range(self.n_strat):
+            # how to handle strata w/ only one cluster?
+            w[self.strat == s] *= float(self.clust_per_strat[s] - 1) \
+                                         / self.clust_per_strat[s]
+            # If there is only one cluster then weights wont change
+            if self.clust_per_strat[s] == 1:
+                continue
+
+            # resample array of clusters
+            ii_resample = np.random.choice(self.ii[s], size=(self.clust_per_strat[s]-1))
+            # accumulate number of times cluster i was resampled
+            clust_count += np.bincount(ii_resample,
+                                       minlength=max(self.sclust)+1)
+
+        w *= clust_count[self.sclust]
+        return w
+
+    def _mean_bootstrap_weight(self, bsn):
+        """
+        Computes mean bootstrap replicate weight
+
+        Parameters
+        ----------
+        bsn : integer
+            Mean bootstrap averages the number of resampled clusters over bsn
+        Returns
+        -------
+        w : ndarray
+            Augmented weight
+        """
+        clust_count = np.zeros(self.design.nclust)
+        w = self.weights.copy()
+        # for each replicate, I accumulate bsn number of times?
+        for b in range(bsn):
+            for s in range(self.n_strat):
+                w[self.strat == s] *= ((float(self.clust_per_strat[s] - 1) /
+                                        self.clust_per_strat[s])**(1/bsn))
+                # If there is only one or two clusters then weights wont change
+                if self.clust_per_strat[s] in (1, 2):
+                    continue
+                # resample array of clusters in strata s
+                ii_resample = np.random.choice(self.ii[s], size=(self.clust_per_strat[s]-1))
+                # accumulate number of times cluster i was resampled
+                clust_count += np.bincount(ii_resample,
+                                           minlength=max(self.sclust)+1)
+        # avg number of times cluster i was resampled
+        clust_count /= bsn
+        # augment weights
+        w *= clust_count[self.sclust]
+        return w
+
+
+class SurveyStat(object):
+    """
+    Estimation and inference for summary statistics in complex surveys.
+
+    Parameters
+    -------
+    design : SurveyDesign object
+
+    Attributes
+    ----------
+    est : ndarray
+        The point estimates of the statistic, calculated on the columns
+        of data.
+    vcov : ndarray
+        The variance-covariance of the estimates.
+    pseudo : ndarray
+        The jackknife pseudo-values.
+    """
+
+    def __init__(self, design, center_by="stratum"):
+        self.design = design
+        self.center_by = center_by
```

Comment:
Here is a good place to check the validity of center_by (raise if it does not hold a valid value).